Sources/FTPropertyWrappers/Keychain - Support/KeychainCoding.swift

```diff
@@ -0,0 +1,371 @@
+        case is Float80.Type:
+            fatalError("Encoding root type Float80 not supported!")
+        case is Double.Type:
+            fatalError("Encoding root type Double not supported!")
+        case is Bool.Type:
+            return (try decode(from: data) as Bool) as! T
+        case is String.Type:
+            return (try decode(from: data) as String) as! T
+        case is URL.Type:
+            return (try decode(from: data) as URL) as! T
+        case is Data.Type:
+            return (try decode(from: data) as Data) as! T
+        default:
+            return try decoder.decode(T.self, from: data)
+        }
+
+    }
+
+    // MARK: - Encoders overloaded
+    private func encode(_ value: Int) throws -> Data {
+        var data = Int64(value)
+        return Data(bytes: &data, count: MemoryLayout<Int64>.size)
+    }
+
+    private func encode(_ value: Int8) throws -> Data {
+        var data = Int64(value)
+        return Data(bytes: &data, count: MemoryLayout<Int64>.size)
+    }
+
+    private func encode(_ value: Int16) throws -> Data {
+        var data = Int64(value)
+        return Data(bytes: &data, count: MemoryLayout<Int64>.size)
+    }
+
+    private func encode(_ value: Int32) throws -> Data {
+        var data = Int64(value)
+        return Data(bytes: &data, count: MemoryLayout<Int64>.size)
+    }
+
+    private func encode(_ value: Int64) throws -> Data {
+        var data = value
+        return Data(bytes: &data, count: MemoryLayout<Int64>.size)
+    }
+
+    private func encode(_ value: UInt) throws -> Data {
+        var data = UInt64(value)
+        return Data(bytes: &data, count: MemoryLayout<UInt64>.size)
+    }
+
+    private func encode(_ value: UInt8) throws -> Data {
+        var data = UInt64(value)
+        return Data(bytes: &data, count: MemoryLayout<UInt64>.size)
+    }
+
+    private func encode(_ value: UInt16) throws -> Data {
+        var data = UInt64(value)
+        return Data(bytes: &data, count: MemoryLayout<UInt64>.size)
+    }
+
+    private func encode(_ value: UInt32) throws -> Data {
+        var data = UInt64(value)
+        return Data(bytes: &data, count: MemoryLayout<UInt64>.size)
+    }
+
+    private func encode(_ value: UInt64) throws -> Data {
+        var data = value
+        return Data(bytes: &data, count: MemoryLayout<UInt64>.size)
+    }
+
+    private func encode(_ value: Bool) throws -> Data {
+        return try encode( value ? 1 : 0)
+    }
+
+    private func encode(_ value: String) throws -> Data {
+        guard let data = value.data(using: .utf8) else {
+            throw KeychainError.generalEncodingFailure
+        }
+        return data
+    }
+
+    private func encode(_ value: URL) throws -> Data {
+        return try encode(value.absoluteString)
+    }
+
+    private func encode(_ value: Data) throws -> Data {
+        return value
+    }
+
+    // MARK: - Decoders overloaded
+
+    private func decode(from data: Data) throws -> Int {
```

Comment:
What if we used the same signature for decoding as a `SingleValuedecodingContainer`? There is a reason behind it, because you are helping the type system. And if we want to support standard-library coding we will have less work to do.

https://developer.apple.com/documentation/swift/singlevaluedecodingcontainer